### PR TITLE
#633 Change flow create offer with onchain

### PR DESCRIPTION
**Pull request by vietdangquoc**
# Ticket
#633 

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: dong-service/scheduler/sync_offers_job.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+	"dong-service/logger"
+	"dong-service/repository"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+)
+
+const (
+	transactionLookbackHours = 24
```
Comment (duongquang18it3) [inline dong-service/scheduler/sync_offers_job.go]: transactionLookbackHours = 24 em không sure cái hardcode này lắm vì nếu lỡ server die, job die khi khởi động lại, nó sẽ bỏ qua tất cả các giao dịch cũ hơn 24h
Em nghĩ nên dùng  logic dựa trên created_at của Offer. Chỉ quét các giao dịch có created_at (transaction) >= created_at (offer) trừ đi một khoảng buffer nhỏ (ví dụ 5 phút).
Thay vì lấy một mốc giờ cố định cho tất cả, mình nên lấy mốc giờ riêng biệt cho từng Offer, nên mình có thể:
- Thêm trường CreatedAt vào PendingOffer
- Cập nhật query lấy Offer: Lấy thêm cột created_at từ DB
- Thay đổi cách tính lookbackTime trong  hàm processOffer kiểu như: lookbackTime := offer.CreatedAt.Add(-5 * time.Minute)

Cái này em nghĩ nó đỡ phần nào cho việc thay vì quét toàn bộ giao dịch của 24h (có thể lên tới hàng triệu record), Job sẽ chỉ quét các giao dịch trong khoảng thời gian hẹp hơn (từ lúc Offer sinh ra đến nay)

File: dong-service/config/config.go [hunk 2]
```diff
@@ -82,6 +82,7 @@ type BlockchainConfig struct {
 
 type SchedulerConfig struct {
 	SyncContributorsInterval    int `mapstructure:"sync_contributors_interval"`     // in seconds
+	SyncOffersInterval          int `mapstructure:"sync_offers_interval"`           // in seconds
```
Comment (hoangdohuy) [inline dong-service/config/config.go]: Cần không?

File: dong-service/handlers/offer_handlers.go [hunk 3]
```diff
@@ -190,22 +190,15 @@ func (h *OfferHandler) ListOffers(c *gin.Context) {
...
 		"success": true,
 		"message": "Offers retrieved",
 		"data":    offers,
 		"meta": gin.H{
-			"page":            pg.Page + 1,
-			"limit":           pg.Limit,
-			"total_items":     total,
-			"total_pages":     totalPage,
-			"total_available": totalAvailable,
+			"page":        pg.Page + 1,
```
Comment (hoangdohuy) [inline dong-service/handlers/offer_handlers.go]: Sao lại code change?

File: indexer/internal/storage/postgres.go [hunk 4]
```diff
@@ -1554,6 +1555,9 @@ func (p *PostgresConnector) insertTransactionsTx(
 	for i := range transactions {
 		t := &transactions[i]
 		t.TransactionExtraInfoType = detectTransactionType(t.ExtraInfo)
+		if t.TransactionExtraInfoType == common.TransactionExtraInfoP2PTrading && t.ExtraInfo != "" {
+			p.updateOfferStatus(ctx, tx, t)
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: không đi hit db trong vòng loop như thế
tham khảo `insertUserContentsTx`

File: indexer/internal/storage/postgres.go [hunk 5]
```diff
@@ -2360,3 +2364,47 @@ func (p *PostgresConnector) GetAllTransactionsByWallet(
...
+	if extra.UserSenderId != nil && offerSellerUserId != "" && *extra.UserSenderId != offerSellerUserId {
+		log.Error().Int64("offer_id", offerID).Str("expected_user_id", offerSellerUserId).Str("actual_user_id", *extra.UserSenderId).Msg("Transaction user id mismatch for offer update")
+		return
+	}
+
+	_, err = tx.ExecContext(ctx, `UPDATE offers SET status = $1, transaction_hash = $2, updated_at = NOW() WHERE offer_id = $3 AND status IN ('OPEN', 'PENDING')`, status, txHash, offerID)
+	if err != nil {
+		log.Error().Err(err).Int64("offer_id", offerID).Str("status", status).Str("tx_hash", txHash).Msg("Failed to update offer status from p2p-trading tx")
+	} else {
+		log.Info().Int64("offer_id", offerID).Str("status", status).Str("tx_hash", txHash).Msg("Updated offer status from p2p-trading tx")
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: socket bắn ở đâu?

File: frontend/modules/p2p/components/p2p-trading/create-offer-form/create-offer-modal.tsx [hunk 6]
```diff
@@ -117,30 +115,17 @@ export const CreateOfferModal = () => {
...
       if (transferResult.success) {
-        setTimeout(() => {
-          updateOfferStatus({
-            offer_id: Number(resultData.offer.offer_id),
-            tx_hash: transferResult.txHash!,
-            status: 'CONFIRMED',
-          });
-        }, 2000);
-        toast.success('Create offer success!');
+        toast.success('Create offer success! Status will be updated automatically.');
```
Comment (hoangdohuy) [inline frontend/modules/p2p/components/p2p-trading/create-offer-form/create-offer-modal.tsx]: Offer của bạn đang được xử lý vui lòng chờ trong giây lát kiểu thế, hỏi AI translate sang TA

File: indexer/internal/storage/postgres.go [hunk 7]
```diff
@@ -2360,3 +2385,183 @@ func (p *PostgresConnector) GetAllTransactionsByWallet(
...
+
+		var (
+			dbFrom   sql.NullString
+			dbTo     sql.NullString
+			dbValue  sql.NullString
+			dbStatus sql.NullInt64
+		)
+
+		if err := tx.QueryRowContext(ctx, `
+			SELECT from_address, to_address, value, status
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: sao lại cẩn kiểm tra?

File: indexer/internal/storage/postgres.go [hunk 8]
```diff
@@ -805,6 +806,30 @@ func (p *PostgresConnector) insertBlockAndTransactions(ctx context.Context, bloc
 		if err != nil {
 			return err
 		}
+
+		var offerList []common.OfferList
+		for i := range blockData.Transactions {
+			tx := &blockData.Transactions[i]
+			tx.TransactionExtraInfoType = detectTransactionType(tx.ExtraInfo)
+			if tx.TransactionExtraInfoType == common.TransactionExtraInfoP2PTrading && tx.ExtraInfo != "" {
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: filter luôn theo status đi chỉ lấy confirmed vs finalized

File: indexer/internal/storage/postgres.go [hunk 9]
```diff
@@ -2360,3 +2385,183 @@ func (p *PostgresConnector) GetAllTransactionsByWallet(
...
+			}
+			continue
+		}
+
+		if !dbStatus.Valid || (dbStatus.Int64 != int64(pb.TransactionStatus_CONFIRMED) && dbStatus.Int64 != int64(pb.TransactionStatus_FINALIZED)) {
+			log.Debug().Int64("offer_id", oid).Int64("tx_status", dbStatus.Int64).Str("skip_reason", "tx_status_invalid").Msg("Transaction status not acceptable; skipping")
+			continue
+		}
+
+		if !dbFrom.Valid || dbFrom.String != dbO.seller {
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: validate seller -> oke nhưng dùng map transaction data đẩy vào thay vì query

File: indexer/internal/storage/postgres.go [hunk 10]
```diff
@@ -805,6 +806,30 @@ func (p *PostgresConnector) insertBlockAndTransactions(ctx context.Context, bloc
...
+				var extra P2PExtraInfo
+				if err := json.Unmarshal([]byte(t.ExtraInfo), &extra); err == nil && extra.OfferID != 0 {
+					offerIDMap[t.Hash] = extra.OfferID
+				}
+			}
+		}
+
+		err = p.updateOfferStatus(ctx, tx, txMap, offerIDMap)
+		if err != nil {
+			return err
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: logger.error
